### ISSUE TITLE
Verbose flag for extra log levels when troubleshooting Astro CLI issues

### ISSUE THREAD

**Issue by ben-astro** [astronomer/astro-cli#653]
- [ X] I have checked that a similar [feature request](https://github.com/astronomer/astro-cli/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22) does not already exist.

**✍️ Is your feature request related to a problem? Please describe.**

This specific issue revolved around troubleshooting `astro dev start` hanging forever, on the pip installs of a requirements.txt file. Would like to know which packages are breaking/failing/hanging, so I don't have to go in and test each package or group of packages independently. 

I have also had various other CLI problems in the past that I think would benefit from additional logging level capabilities, like which part of `astro deploy` is failing. 

**🧩 Describe the solution you'd like**

I'd like to have a simple `logging verbose` or `verbosity debug` flag that will print out additional information about individual actions/steps the CLI is implementing, to aid in troubleshooting issues. I realize this is rather general and vague, but basically a toggle that can print out extra info and help us pinpoint issues like which packages are failing to install, or allow us to get more info about what exact step is breaking when running various CLI commands, printing out what's happening when CLI is authenticating to Astro Cloud, etc etc. 

**🤔 Describe alternatives you've considered**

The alternative is NOT having this extra info, which can be worked around but can be frustrating when troubleshooting issues. 

**💡 Additional context**

It's relatively uncommon that I have run into CLI issues that would benefit from this level of logging, so definitely not HIGH priority, but one of those things that's super nice to have in place when those issues occur.

**Comment by danalogalbo-astronomer** [comment]
@neel-astro or @feluelle I see this is moved from `done` to new - is this something we do want to keep?

**Comment by neel-astro** [comment]
> [@neel-astro](https://github.com/neel-astro) or [@feluelle](https://github.com/feluelle) I see this is moved from `done` to new - is this something we do want to keep?

Yes, this has been an ask in the Slack channel a bunch of times over the years, and it is something that the Astro side of Astro CLI is lacking and Software/APC already has.
